app/controllers/member_conversion_controller.rb

```diff
@@ -0,0 +1,202 @@
+class MemberConversionController < ApplicationController
+  include CurationConcerns::Lockable
+
+  #Promote a FileSet to a child GenericWork.
+  def to_child_work
+    begin
+      parent_work = GenericWork.find(params['parentid'])
+      file_set = FileSet.find(params['filesetid'])
+    rescue
+      flash[:notice] = "Sorry. This item no longer exists."
+      redirect_to "/works/#{parent_work.id}"
+      return
+    end
+
+    if !validate_for_switch_to_child_work(parent_work, file_set)
+      flash[:notice] = "\"#{file_set.title.first}\" can't be promoted to a child work."
+      redirect_to "/concern/parent/#{parent_work.id}/file_sets/#{file_set.id}"
+      return
+    end
+    place_in_order = parent_work.ordered_members.to_a.find_index(file_set)
+    transfer_thumbnail = (parent_work.thumbnail == file_set)
+    transfer_representative = (parent_work.representative == file_set)
+    new_child_work = GenericWork.new(title: file_set.title)
+    acquire_lock_for(parent_work.id) do
+        new_child_work.apply_depositor_metadata(current_user.user_key)
+        new_child_work.creator = [current_user.user_key]
+        new_child_work.save!
+        copy_metadata_from_parent(parent_work, new_child_work)
+        parent_work.representative_id = nil if transfer_representative
+        parent_work.thumbnail_id = nil if transfer_thumbnail
+        remove_member_from_parent(parent_work, file_set)
+        parent_work.save!
+        add_member_to_parent(new_child_work, file_set, 0)
+        set_thumbnail_and_rep(new_child_work, file_set, true, true)
+        set_thumbnail_and_rep(parent_work, new_child_work, transfer_thumbnail, transfer_representative)
+        new_child_work.save!
```

Comment:
Also not sure if there's a way to re-organize these so you don't need to save the parent work and child work _twice_ each -- we know saves are expensive. Can it be reordered so each only needs to be saved once?

There's a lot of things going on here -- I think they all really need to go  on -- but maybe re-organizing in into "paragraphs" with a little comment header explaining what's going on would help. Not totally sure. I don't totally follow the `transfer_representative` stuff though, maybe a brief comment explaining why you only sometimes need to do this.